arch/arm/armv7-m/include/arch_helpers.h

```diff
@@ -8,15 +8,29 @@
 #ifndef ARCH_HELPERS_H
 #define ARCH_HELPERS_H
 
+/*
+ * This variable is used to ensure spurious nested calls won't
+ * enable interrupts. This is been defined in arch_main.c
+ */
+extern unsigned int critical_section_nest_level;
+
 /*!
  * \brief Enables global CPU interrupts.
  *
  * \note inline is necessary as this call can be used in performance sensitive
  *     path
  */
-inline static void arch_interrupts_enable(void)
+inline static void arch_interrupts_enable(unsigned int not_used)
 {
-    __asm volatile("cpsie i" : : : "memory");
+   /* Decrement critical_section_nest_level only if in critical section */
+    if (critical_section_nest_level > 0) {
+        critical_section_nest_level--;
+    }
+
+    /* Enable interrupts globally if now outside critical section */
+    if (critical_section_nest_level == 0) {
+        __asm volatile("cpsie i" : : : "memory");
```

Comment:
Using asm here is triggering MISRA violation. I understand this is something introduced by this patch but we have to resolve it.
I suggest using the CMSIS equivalent. Can you please replace `__asm volatile("cpsie i" : : : "memory");` with `__enable_irq();` and `__asm volatile("cpsid i" : : : "memory");` with `__disable_irq()` so we can pass this check?